src/Auth0Client.ts

```diff
@@ -498,7 +498,10 @@ export default class Auth0Client {
       delete options.client_id;
     }
     ClientStorage.remove('auth0.is.authenticated');
-    const { federated, ...logoutOptions } = options;
+    const { federated, localOnly, ...logoutOptions } = options;
+    if (localOnly) {
+      return;
+    }
```

Comment:
Just wondering if we need to do something in the scenario where the developer specifies both federated: true and localOnly: true in the options. In the former, they’re saying they want to log out from everywhere, not just Auth0 (so that could include the social IdP as well) and in the latter they’re saying they only want to log out of the local app. They feel conflicting.
I’ve seen other libraries throw an error in these types of cases (not necessarily Auth0 ones). If we don’t do anything, how best to let the developer know which one takes precedence?

Reply:
Yeah, I suppose specifying any `logoutOptions` in addition to `localOnly` doesn't make sense.

Perhaps a new method?

```js
logoutLocally() {
  ClientStorage.remove('auth0.is.authenticated');
}

logout(options) {
    this.logoutLocally();
    // logout remotely
}
```

Also, we might want to do `window.reload()` else logging out locally wont actually change the logged in state of the app.

Reply:
I'm not sure about the additional method yet, I think just having one logout method would keep the API simple. What do you think @damieng?

However, the `reload` thing is interesting. Right now we'd expect the application to do that redirect if they need to, or we can handle that in the React hook (as an example) where we can change app state that way. For a SPA a full-page reload would probably not be a desirable side-effect and I can see tickets being raised asking us to add options to disable that as well.